projects/plugins/super-cache/wp-cache.php

```diff
@@ -1505,6 +1505,7 @@ function wpsc_update_direct_pages() {
 				$out .= "'$page', ";
 			}
 		}
+		// @phan-suppress-next-line PhanRedundantCondition -- $cached_direct_pages can be false if unset and the above loop never runs
```

Comment:
The loop should always run thanks to the `is_array( $_POST[ 'direct_pages' ] ) && !empty( $_POST[ 'direct_pages' ] )` checks on line 1498. OTOH, `$cached_direct_pages` may still be unset if `$page` winds up as the empty string on every iteration.

But all this seems unnecessarily confusing. Why not just set it to an empty array on line 1500 in the first place?

Reply:
Good thought: 75fa5ad139c

That also prevents us from setting an array element on a potentially undefined var, too.